doc/system_specific_setup.rst

```diff
@@ -1,225 +1,266 @@
+Installing LaTeX on Ubuntu
+++++++++++++++++++++++++++
+The following three packages need to be installed to create the pdf documentation.
+It is recommended to install the packages in one ``apt-get`` command, since
+otherwise depending on filesystem unnecessary snapshots might be created that
+take up additional memory. The ``--no-install-recommends`` option significantly
+reduces the number of installed packages::
 
-::
+   > sudo apt-get install --no-install-recommends texlive \
+   texlive-latex-extra latexmk
 
-   > sudo apt-get install texlive
-   > sudo apt-get install texlive-latex-extra
+Installing LaTeX on OpenSUSE
+++++++++++++++++++++++++++++
+The following command installs the minimum number of packages
+in order to create the pdf documentation. The following command
+installs around 130 packages all in all, requiring around 300 MB.
 
-You can now build the pdf documentation
-::
 
-   > cd doc
-   > make latexpdf
+.. warning::
 
-Static code tests and style checking
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    It is important to install the packages in one ``zypper`` command, since
+    otherwise depending on used file system snapshots might be created after
+    each package, which can add up to several GB of data.
 
-Install the standalone pep8 tool
 ::
 
-   > sudo apt-get install pep8
+   > sudo zypper install --no-recommends texlive-latex texlive-latexmk \
+   texlive-babel-english texlive-cmap texlive-psnfss texlive-fncychap  \
+   texlive-fancyhdr texlive-titlesec texlive-tabulary texlive-varwidth \
+   texlive-framed texlive-fancyvrb texlive-float texlive-wrapfig       \
+   texlive-parskip texlive-upquote texlive-capt-of texlive-needspace   \
+   texlive-metafont texlive-makeindex texlive-times texlive-helvetic   \
+   texlive-courier texlive-dvips
 
-You can now test whether the Python code conforms to the pep8
-standards
-::
 
-   > pep8 code.py
+Creating pdf documentation
+++++++++++++++++++++++++++
 
-Install the standalone pylint tool
+You can now build the pdf documentation using
 ::
 
-   > sudo apt-get install pylint
+   > cd doc
+   > make latexpdf
 
-You can now test how well the Python code conforms to the pylint
-standards
-::
+Installing testing tools
+^^^^^^^^^^^^^^^^^^^^^^^^
+The following modules are recommended to get access to testing and
+formatting tools::
 
-   > pylint code.py
+   > sudo pip install pytest pep8 pylint pytest-cov pytest-pep8 \
+   pytest-pylint pytest-flakes pytest-pep257
 
-Finally, install useful pytest extensions using pip:
-::
 
-   > sudo pip install pytest-cov
-   > sudo pip install pytest-pep8
-   > sudo pip install pytest-pylint
-   > sudo pip install pytest-flakes
-   > sudo pip install pytest-pep257
+You can now run the PSyclone python tests::
 
-If you don't have root access then you can specify the ``--user`` argument to
-install packages in a user-local directory -- see the instructions on
-:ref:`install_fparser` above.
+   > cd PSyclone.git/src/psyclone/tests
```

Comment:
If we've installed via pip then you only need to `cd PSyclone.git` - the tests will/should run from the top-level directory.